Integration tests race their own 10-second voting deadline

Four meal planning integration tests give a meal plan a **10-second** voting deadline and then
spend an unbounded amount of wall clock casting votes against it. Under a loaded package run the
setup outruns the deadline, a parallel test's worker sweeps the plan up and finalizes it, and the
remaining votes are refused.

## The failure

```
--- FAIL: TestMealPlanTasks_UpdateMealPlanTaskStatus/happy_path (33.98s)
    mealplanning_meal_plan_tasks_test.go:165:
        Error: Received unexpected error:
               rpc error: code = FailedPrecondition desc = meal plan event is not eligible
               for voting: meal plan event da32v9cn9qde2kqfrtug is not eligible for voting
```

Seen 2026-08-19 on a full `scripts/integration_tests_postgres.sh` run (the apiserver package took
137s; this subtest alone took 34s). Passes in isolation — `go test -run
TestMealPlanTasks_UpdateMealPlanTaskStatus ./testing/integration/apiserver/` is green in 13s.

## Why

Eligibility is not about the deadline. `MealPlanEventIsEligibleForVoting`
(`internal/repositories/postgres/mealplanning/sqlc_queries/meal_plan_events.generated.sql:41`)
only asks whether `meal_plans.status = 'awaiting_votes'`. So the error means the plan's status
left `awaiting_votes` **while the vote loop was still running**.

What moves it is `GetMealPlansAwaitingFinalizationSaga`
(`.../sqlc_queries/meal_plans.generated.sql:65`):

```sql
WHERE meal_plans.archived_at IS NULL
  AND meal_plans.finalization_saga_id IS NULL
  AND ((meal_plans.status = 'awaiting_votes' AND meal_plans.voting_deadline < NOW()) OR ...)
```

That predicate is database-wide — it is not scoped to an account, a test, or a plan. So the race
is:

1. `createFinalizedMealPlanWithTasks` sets `VotingDeadline: now.Add(10 * time.Second)`
   (`mealplanning_meal_plan_tasks_test.go:132`).
2. The vote loop runs three users through `CreateMealPlanOptionVote`. Under load that takes
   longer than ten seconds.
3. The deadline passes mid-loop. Any parallel test in the package that calls
   `RunMealPlanTaskCreatorWorker` or `RunFinalizeMealPlanWorker` — several do — now matches this
   plan, attaches a saga, and the saga worker finalizes it.
4. The next `CreateMealPlanOptionVote` in the loop gets `FailedPrecondition`.

The helper's own comment shows the neighbouring hazard was already known — *"Finalize the meal
plan directly (per-plan, not global worker, to avoid conflicts with parallel tests)"* — but the
deadline itself is still a wall-clock budget on test setup, and it is the shortest one in the
suite by a factor of thirty.

## Sites

Every one of these creates a plan in `awaiting_votes` with a ten-second deadline and then does
real work before or during voting:

- `mealplanning_meal_plan_tasks_test.go:132` (`createFinalizedMealPlanWithTasks`, feeding
  `TestMealPlanTasks_UpdateMealPlanTaskStatus` and `TestMealPlanTasks_RunMealPlanTaskCreatorWorker`)
- `mealplanning_meal_plan_recipe_option_selections_test.go:106`
  (`TestMealPlans_WithRecipeOptionSelections`)
- `mealplanning_meal_plans_test.go:515`
- `mealplanning_meal_plans_test.go:248` (`const baseDeadline = 10 * time.Second`, used at :254)

The rest of the suite already uses five minutes or an hour, and those have not been seen to fail
this way.

## Suggested fix

Raise the four deadlines to something no amount of load can outrun — an hour, matching
`mealplanning_meal_plan_recipe_option_selections_test.go:550` and `mealplanning_recipes_test.go:937`.
Nothing is lost: every one of these tests already forces the deadline into the past explicitly
after voting (`VotingDeadline: time.Now().Add(-time.Minute)`) and then finalizes the plan
directly. The short deadline is not what the tests are asserting on; it is only what makes them
race.

Worth considering alongside it: the reason a *different* test's plan can be finalized at all is
that the worker endpoints are global. Scoping `GetMealPlansAwaitingFinalizationSaga` per call, or
having the suite stop calling the global starter where a direct `FinalizeMealPlan` would do,
would remove the class rather than this instance of it. That is a larger change and probably a
separate ticket.

## Notes

`TestMealPlans_WithRecipeOptionSelections` also has a *second*, unrelated intermittent failure —
`GetAllMealPlanOptionsForMealPlanEvent` has no `ORDER BY`, so the test's positional
`Options[0]` assumption elects the wrong meal when Postgres returns the rows the other way round
(seen 2026-08-18 in #1323's CI). Same test, different root cause; fixing the deadline will not
fix that one.

--- a/backend/cmd/tools/codegen/queries/mealplanning_meal_plan_events.go
+++ b/backend/cmd/tools/codegen/queries/mealplanning_meal_plan_events.go
@@ -129,13 +129,15 @@ GROUP BY %s.%s
 FROM %s
 WHERE
 	%s.%s IS NULL
-	AND %s.%s = sqlc.arg(%s);`,
+	AND %s.%s = sqlc.arg(%s)
+ORDER BY %s.%s ASC;`,
 						strings.Join(applyToEach(mealPlanEventsColumns, func(i int, s string) string {
 							return fmt.Sprintf("%s.%s", mealPlanEventsTableName, s)
 						}), ",\n\t"),
 						mealPlanEventsTableName,
 						mealPlanEventsTableName, archivedAtColumn,
 						mealPlanEventsTableName, belongsToMealPlanColumn, mealPlanIDColumn,
+						mealPlanEventsTableName, idColumn,
 					)),
 				},
 				{

--- a/backend/cmd/tools/codegen/queries/mealplanning_meal_plan_options.go
+++ b/backend/cmd/tools/codegen/queries/mealplanning_meal_plan_options.go
@@ -158,7 +158,8 @@ WHERE
 	AND meal_plan_events.id = sqlc.arg(meal_plan_event_id)
 	AND meal_plan_events.belongs_to_meal_plan = sqlc.arg(meal_plan_id)
 	AND meal_plans.archived_at IS NULL
-	AND meal_plans.id = sqlc.arg(meal_plan_id);`,
+	AND meal_plans.id = sqlc.arg(meal_plan_id)
+ORDER BY meal_plan_options.id ASC;`,
 						strings.Join(fullSelectColumns, ",\n\t"),
 						mealPlanOptionsTableName,
 					)),

--- a/backend/internal/repositories/postgres/mealplanning/generated/meal_plan_events.generated.sql_generated.go
+++ b/backend/internal/repositories/postgres/mealplanning/generated/meal_plan_events.generated.sql_generated.go
@@ -108,6 +108,7 @@ FROM meal_plan_events
 WHERE
 	meal_plan_events.archived_at IS NULL
 	AND meal_plan_events.belongs_to_meal_plan = $1
+ORDER BY meal_plan_events.id ASC
 `
 
 func (q *Queries) GetAllMealPlanEventsForMealPlan(ctx context.Context, db DBTX, mealPlanID string) ([]*MealPlanEvents, error) {

--- a/backend/internal/repositories/postgres/mealplanning/generated/meal_plan_options.generated.sql_generated.go
+++ b/backend/internal/repositories/postgres/mealplanning/generated/meal_plan_options.generated.sql_generated.go
@@ -212,6 +212,7 @@ WHERE
 	AND meal_plan_events.belongs_to_meal_plan = $2
 	AND meal_plans.archived_at IS NULL
 	AND meal_plans.id = $2
+ORDER BY meal_plan_options.id ASC
 `
 
 type GetAllMealPlanOptionsForMealPlanEventParams struct {

--- a/backend/internal/repositories/postgres/mealplanning/sqlc_queries/meal_plan_events.generated.sql
+++ b/backend/internal/repositories/postgres/mealplanning/sqlc_queries/meal_plan_events.generated.sql
@@ -126,7 +126,8 @@ SELECT
 FROM meal_plan_events
 WHERE
 	meal_plan_events.archived_at IS NULL
-	AND meal_plan_events.belongs_to_meal_plan = sqlc.arg(meal_plan_id);
+	AND meal_plan_events.belongs_to_meal_plan = sqlc.arg(meal_plan_id)
+ORDER BY meal_plan_events.id ASC;
 
 -- name: UpdateMealPlanEvent :execrows
 UPDATE meal_plan_events SET

--- a/backend/internal/repositories/postgres/mealplanning/sqlc_queries/meal_plan_options.generated.sql
+++ b/backend/internal/repositories/postgres/mealplanning/sqlc_queries/meal_plan_options.generated.sql
@@ -106,7 +106,8 @@ WHERE
 	AND meal_plan_events.id = sqlc.arg(meal_plan_event_id)
 	AND meal_plan_events.belongs_to_meal_plan = sqlc.arg(meal_plan_id)
 	AND meal_plans.archived_at IS NULL
-	AND meal_plans.id = sqlc.arg(meal_plan_id);
+	AND meal_plans.id = sqlc.arg(meal_plan_id)
+ORDER BY meal_plan_options.id ASC;
 
 -- name: GetMealPlanOptions :many
 SELECT

--- a/backend/testing/integration/apiserver/mealplanning_finalization_helpers.go
+++ b/backend/testing/integration/apiserver/mealplanning_finalization_helpers.go
@@ -25,6 +25,20 @@ const (
 	finalizationPollInterval = 100 * time.Millisecond
 )
 
+// votingDeadlineForSetup is the deadline a test gives a plan it intends to vote on itself.
+//
+// The deadline is a wall-clock budget on everything the test does between creating the plan and
+// casting its last ballot, because the predicate behind GetMealPlansAwaitingFinalizationSaga is
+// database-wide: once a plan's deadline passes while it is still awaiting votes, the next
+// RunFinalizeMealPlanWorker or RunMealPlanTaskCreatorWorker call from *any* parallel test in the
+// package sweeps it up, attaches a saga, and finalizes it. The remaining votes are then refused
+// with "meal plan event is not eligible for voting".
+//
+// An hour is not a wait — no test ever reaches it. It is long enough that no amount of load
+// makes the deadline the thing that fails. Tests that need an expired deadline set one
+// explicitly once they are done voting, which is what actually drives finalization.
+const votingDeadlineForSetup = time.Hour
+
 // awaitMealPlanFinalized waits for the finalization saga's first step to tally a meal plan's
 // ballots and mark it finalized, and returns the plan as it stands afterwards.
 func awaitMealPlanFinalized(t *testing.T, ctx context.Context, c client.Client, mealPlanID string) *mealplanning.MealPlan {

--- a/backend/testing/integration/apiserver/mealplanning_meal_plan_recipe_option_selections_test.go
+++ b/backend/testing/integration/apiserver/mealplanning_meal_plan_recipe_option_selections_test.go
@@ -103,7 +103,7 @@ func TestMealPlans_WithRecipeOptionSelections(T *testing.T) {
 		exampleMealPlan := &mealplanning.MealPlan{
 			Notes:          t.Name(),
 			Status:         string(mealplanning.MealPlanStatusAwaitingVotes),
-			VotingDeadline: now.Add(10 * time.Second),
+			VotingDeadline: now.Add(votingDeadlineForSetup),
 			ElectionMethod: mealplanning.MealPlanElectionMethodSchulze,
 			Events: []*mealplanning.MealPlanEvent{
 				{
@@ -547,7 +547,7 @@ func createMealPlanWithAlternativeIngredientsForSelectionTests(t *testing.T) *se
 		Notes:  t.Name(),
 		Status: string(mealplanning.MealPlanStatusFinalized),
 		// voting deadline must be before every event's start time (events start 24h out); see MealPlanCreationRequestInput.ValidateWithContext.
-		VotingDeadline: now.Add(1 * time.Hour),
+		VotingDeadline: now.Add(votingDeadlineForSetup),
 		ElectionMethod: mealplanning.MealPlanElectionMethodSchulze,
 		Events: []*mealplanning.MealPlanEvent{
 			{

--- a/backend/testing/integration/apiserver/mealplanning_meal_plan_tasks_test.go
+++ b/backend/testing/integration/apiserver/mealplanning_meal_plan_tasks_test.go
@@ -129,7 +129,7 @@ func createFinalizedMealPlanWithTasks(t *testing.T) (string, client.Client) {
 	exampleMealPlan := &mealplanning.MealPlan{
 		Notes:          t.Name(),
 		Status:         string(mealplanning.MealPlanStatusAwaitingVotes),
-		VotingDeadline: now.Add(10 * time.Second),
+		VotingDeadline: now.Add(votingDeadlineForSetup),
 		ElectionMethod: mealplanning.MealPlanElectionMethodSchulze,
 		Events: []*mealplanning.MealPlanEvent{
 			{

--- a/backend/testing/integration/apiserver/mealplanning_meal_plans_test.go
+++ b/backend/testing/integration/apiserver/mealplanning_meal_plans_test.go
@@ -245,13 +245,12 @@ func TestMealPlans_CompleteLifecycleForAllVotesReceived(T *testing.T) {
 			createdMeals = append(createdMeals, createdMeal)
 		}
 
-		const baseDeadline = 10 * time.Second
 		now := time.Now()
 
 		exampleMealPlan := &mealplanning.MealPlan{
 			Notes:          t.Name(),
 			Status:         string(mealplanning.MealPlanStatusAwaitingVotes),
-			VotingDeadline: now.Add(baseDeadline),
+			VotingDeadline: now.Add(votingDeadlineForSetup),
 			ElectionMethod: mealplanning.MealPlanElectionMethodSchulze,
 			Events: []*mealplanning.MealPlanEvent{
 				{
@@ -512,7 +511,7 @@ func TestMealPlans_FinalizeMealPlan(T *testing.T) {
 		exampleMealPlan := &mealplanning.MealPlan{
 			Notes:          t.Name(),
 			Status:         string(mealplanning.MealPlanStatusAwaitingVotes),
-			VotingDeadline: now.Add(10 * time.Second),
+			VotingDeadline: now.Add(votingDeadlineForSetup),
 			ElectionMethod: mealplanning.MealPlanElectionMethodSchulze,
 			Events: []*mealplanning.MealPlanEvent{
 				{
@@ -723,13 +722,12 @@ func TestMealPlans_CompleteLifecycleForSomeVotesReceived(T *testing.T) {
 			createdMeals = append(createdMeals, createdMeal)
 		}
 
-		const baseDeadline = time.Minute
 		now := time.Now()
 
 		exampleMealPlan := &mealplanning.MealPlan{
 			Notes:          t.Name(),
 			Status:         string(mealplanning.MealPlanStatusAwaitingVotes),
-			VotingDeadline: now.Add(baseDeadline),
+			VotingDeadline: now.Add(votingDeadlineForSetup),
 			ElectionMethod: mealplanning.MealPlanElectionMethodSchulze,
 			Events: []*mealplanning.MealPlanEvent{
 				{

--- a/backend/testing/integration/apiserver/mealplanning_recipes_test.go
+++ b/backend/testing/integration/apiserver/mealplanning_recipes_test.go
@@ -934,7 +934,7 @@ func TestRecipes_GetMealPlanTasksForRecipe(T *testing.T) {
 			GroceryListInitialized: true,
 			TasksCreated:           true,
 			// voting deadline must be before every event's start time; see MealPlanCreationRequestInput.ValidateWithContext.
-			VotingDeadline: now.Add(1 * time.Hour),
+			VotingDeadline: now.Add(votingDeadlineForSetup),
 			ElectionMethod: mealplanning.MealPlanElectionMethodSchulze,
 			Events: []*mealplanning.MealPlanEvent{
 				{

